src/components/quiz/index.tsx

```diff
@@ -0,0 +1,156 @@
+"use client";
+import { useState, useEffect } from "react";
+import Image from "next/image";
+import dynamic from "next/dynamic";
+import { cn } from "@/lib/utils";
+import QuizButton from "./quiz-button";
+import useMediaQuery from "@/hooks/use-media-query";
+import { Button } from "@/components/ui/button";
+import Caret from "@/icons/caret-right.svg";
+import AnsweredAnimation from "./answered-animation";
+
+const AnsweredAnimationMobile = dynamic(() => import("./answered-animation-mobile"), {
+  ssr: false,
+});
+
+export interface QuizData {
+  key: string;
+  colorClass: string;
+  chapterNumber: string;
+  chapterName: string;
+  questions: {
+    title: React.ReactNode;
+    options: string[];
+    answer: number;
+    images?: string[];
+  }[];
+}
+
+const Quiz = ({ data }: { data: QuizData }) => {
+  const [hoveredOption, setHoveredOption] = useState<number | null>(null);
+  const [selectedOption, setSelectedOption] = useState<number | null>(null);
+  const [quizQuestion, setQuizQuestion] = useState<number>(0);
+  const [waitForNext, setWaitForNext] = useState<boolean>(false);
+  const [playMobileAnimation, setPlayMobileAnimation] = useState<boolean>(false);
+
+  const { key, colorClass, chapterNumber, chapterName, questions } = data;
+  const question = questions[quizQuestion];
+  const { title, options, images, answer } = question;
+  const isMobile = useMediaQuery("not (min-width: 1024px)", false);
+
+  useEffect(() => {
+    let timeoutId: NodeJS.Timeout;
+    if (isMobile && playMobileAnimation) {
+      timeoutId = setTimeout(() => {
+        setPlayMobileAnimation(false);
+      }, 2000);
+    }
+
+    return () => clearTimeout(timeoutId);
+  }, [playMobileAnimation, quizQuestion, isMobile]);
+
+  const handleNext = () => {
+    setSelectedOption(null);
+    setWaitForNext(false);
+    setQuizQuestion(quizQuestion + 1);
+  };
+
+  const isRight = selectedOption === answer;
+  return (
+    <div className="flex w-full max-w-7xl flex-col overflow-hidden bg-black pt-10 lg:mb-20 lg:pt-[77px]">
+      <AnsweredAnimationMobile isRight={isRight} visible={playMobileAnimation} />
+      <div className="relative mb-[15px] flex w-full justify-between">
+        <div className="z-10 mx-6 flex flex-col gap-6 text-white lg:ml-[53px] lg:mr-0 lg:max-w-[835px] lg:gap-8 lg:pb-[73px]">
```

Comment:
I think that when the mobile layout is used, the white line should extend until the end of the container. Don't you think?
![The white line after “Check your Understanding” doesn't reach the end of the container](https://github.com/user-attachments/assets/899cb9df-68e4-4177-a24d-5ebbfe43d14c)